EspressoSystems/HotShot
# [Upgradability] - Avoid tracking the version in task states

Closes #3378.

### This PR: 
* Removes version from task states.
* Adds a helper function to dynamically calculate the version.
* Uses the new function to get the version when it's needed, not relying on the task state.

### This PR does not: 
* Change how the upgrade works except for removing the version field update.

### Key places to review: 
* `fn version` in `crates/types/src/simple_certificate.rs`.
* How `fn version` is called.

### How to test this PR:
* Run with --features example-upgrade and see if the upgrade happens around view 25 for both dependency and non-dependency tasks.
* Upgrade task tests should also pass.
